src/vma/util/sys_vars.cpp

```diff
@@ -993,7 +994,14 @@ void mce_sys_var::get_env_params()
 
 	// handle internal thread affinity - default is CPU-0
 	if ((env_ptr = getenv(SYS_VAR_INTERNAL_THREAD_AFFINITY)) != NULL) {
-		snprintf(internal_thread_affinity_str, sizeof(internal_thread_affinity_str), "%s", env_ptr);
+		int n = snprintf(internal_thread_affinity_str,
+				sizeof(internal_thread_affinity_str) - 1, "%s", env_ptr);
+		if (n < 0) {
+			vlog_printf(VLOG_WARNING,"Failed to process: %s.\n",
+					SYS_VAR_INTERNAL_THREAD_AFFINITY);
+		} else {
+			internal_thread_affinity_str[n] = '\0';
```

Comment:
@igor-ivanov Liran comment is correct. The following line is redundant
```
internal_thread_affinity_str[n] = '\0';
```
since it is part of snprintf